### ISSUE TITLE
Treasure AI — Senior Software Engineer – Machine Learning

### ISSUE THREAD

**Issue by louisinayinde** [louisinayinde/jobs#400]
**Entreprise** : Treasure AI  
**Poste** : Senior Software Engineer – Machine Learning  
**Localisation** : Tokyo, JP  
**Score** : 104 — asia\_relocation (tokyo) · stack +24 (AWS, BigQuery, Python, Spark)  
**Tech** : AWS, BigQuery, Python, Spark  
**Lien ATS** : [japan-dev.com](<https://japan-dev.com/jobs/treasure-data/treasure-data-staff-software-engineer--machine-learning-pdra2y>)  
**Publiée le** : 2026-07-21  
**Source** : japandev  

---

**Description**

Type de contrat : temps plein (FULL\_TIME)

Treasure Data:

At Treasure Data, we’re on a mission to radically simplify how companies use data and AI to create connected customer experiences. Our intelligent customer data platform (CDP) drives revenue growth and operational efficiency across the enterprise to deliver powerful business outcomes.

We are thrilled that Forrester has recognized Treasure Data as a Leader in The Forrester Wave™: Customer Data Platforms For B2C. It's an honor to be acknowledged for our efforts in advancing the CDP industry with cutting-edge AI and real-time capabilities.

Furthermore, Treasure Data employees are enthusiastic, data-driven, and customer-obsessed. We are a team of drivers—self-starters who take initiative, anticipate needs, and proactively jump in to solve problems. Our actions reflect our values of honesty, reliability, openness, and humility.

We are investing heavily in our core platform to support sustained growth in data volume, customer complexity, and reliability expectations.

Meet the team

The Machine Learning (ML) team is dedicated to empowering our customers to make data-driven decisions using insights derived from ML solutions. We are revamping our ML infrastructure to make it more scalable and maintainable, and continue to build ML solutions on top of it to be integrated with our award-winning Customer Data Platform (CDP). By integrating these solutions we aim to reduce the hurdles and enable individuals with limited data science experience, such as marketers, to harness the power of Artificial Intelligence (AI).

As an example of the strategic ML investments, our AI Signals solutions, including Recency Frequency Monetary (RFM), Next Best Product (NBP), Next Best Action (NBA), Customer LifeTime Value (CLTV), have been rolled out, with more ML solutions in the pipeline. The team's goal is to empower every customer to take advantage of AI to drive business value without the need for extensive knowledge of data science. We are committed to driving this multi-year initiative of democratizing AI for every customer.

Your Role

We are seeking strong ML engineers to join the ML team and help us sharpen the ML vision and deliver more solutions to satisfy our customers' needs. You will be working closely with other ML engineers and people from cross-functional teams.

You will be a good fit if you:

- Are a self-driven, organized, and independent individual who proactively takes initiatives, anticipates needs, and solves problems to contribute to delivering values to our customers.

- Have a strong sense of ownership and responsibility to get things done.

- Have a growth mindset; are curious to learn new things and adaptive to changes.

- Know how to navigate ambiguity and thrive in uncertain environments, consistently driving work forward.

- Are passionate about productizing ML products, knowing how to make practical trade-offs when turning ideas into working software.

- Excel in adapting communication styles and simplifying complex technical concepts for diverse audiences, ensuring clear and effective communication.

- Enjoy working in a collaborative work environment with people from diverse backgrounds.

This is an ideal position for those with not only data science and machine learning skills, but also cloud service engineering skills for developing, deploying, and operating these critical ML products.

Key Responsibilities

- Design and develop ML products to be integrated with our CDP.

- Own specific technical areas, drive and execute ML product projects, track progress and mitigate risks by collaborating closely with product managers, UX designers, architects, engineers, and stakeholders from other cross-functional teams.

- Play a key role in defining system architecture for the ML products and implementing specific components to enhance the user experience.

- Design and implement performant, scalable ELT (Extract, Load, Transform) data pipelines, considering an ML model's lifecycle (training and inference).

- Take responsibility for technical problem solving and meeting ML product objectives creatively in ambiguous scenarios.

- Participate in the on-call rotation for production support.

- Drive best practices including ML research methodologies, coding standards, code reviews, source control management, development processes, build processes, testing and release, and operational excellence.

Requirements

- Advanced degree in computer science, data science, machine learning, or related field, or equivalent work experience.

- 6+ years of professional experience in software engineering designing and building ML-driven products, with at least 3 years focused on production ML systems.

- Fundamental knowledge of Data Engineering and extensive experience in developing and deploying ML models, as well as building and maintaining ML pipelines and products.

- Experience in applying scientific method: hypothesis formulation and testing, exploratory data analysis, cross-validation, reproducible research, and structured reporting/documentation, result explanation and presentation.

- Experience designing, deploying, and operating scalable ML systems in production. This includes responsibility for model selection, performance benchmarking, and lifecycle management to solve real-world business problems.

- Proficiency in Python and general ML ecosystem tooling in data processing and modelling (such as NumPy, Pandas, Scikit-learn, PyTorch, etc.).

- Experience in designing and building products using public cloud services such as AWS.

- Excellent verbal and written communication skills in English, and ability to convey research findings and implications to both technical and non-technical audiences.

- Ability to work effectively in cross-functional and distributed teams across different time zones.

Nice To Have

- Experience with security design principles and best practices.

- Experience working with big data technologies such as Hive, Trino, Spark, BigQuery, and Redshift.

- OSS contribution experiences.
